analysis/deciles_charts.py

```diff
@@ -166,12 +172,12 @@ def main():
         if config["tables"]["output"]:
             deciles_table = get_deciles_table(measure_table, config)
             fname = f"deciles_table_{id_}.csv"
-            write_deciles_table(deciles_table, output_dir / fname)
+            write_deciles_table(deciles_table, output_dir, fname)
 
         if config["charts"]["output"]:
             chart = get_deciles_chart(measure_table, config)
             fname = f"deciles_chart_{id_}.png"
-            write_deciles_chart(chart, output_dir / fname)
+            write_deciles_chart(chart, output_dir, fname)
```

Comment:
Hah, ok this is not quite what I meant, sorry didn't explain myself properly.

So, there's no point creating the dir inside the `write_` functions if it's always going to be fixed to `output_dir`? Might as well just create it in `main()`?

My thinking in creating the dir inside the `write_` functions was to use `path.parent` and thus to embed the desired behaviour (parent directory creation) at the lowest level. 

That way, future uses of the `write_()` functions will work even if this action writes to subdirectories of `output_dir` (which it doesn't *currently* do). Basically a more future proof implementation.

However, this is all a bit academic, as it would be a simple fix to change later! So approving as is :)

Reply:
Ah sorry, misunderstood that part of the change. Definitely agree with changing it later at this point.